src/server/pachyderm_test.go

```diff
@@ -9648,6 +9647,56 @@ func TestMoveBranchTrigger(t *testing.T) {
 	require.Equal(t, "/foo", files[0].File.Path)
 }
 
+func TestExpiredFileset(t *testing.T) {
+	if testing.Short() {
+		t.Skip("Skipping integration tests in short mode")
+	}
+
+	c := tu.GetPachClient(t)
+	require.NoError(t, c.DeleteAll())
+
+	dataRepo := tu.UniqueString(t.Name())
+	require.NoError(t, c.CreateRepo(dataRepo))
+
+	// create a pipeline taking both master and the trigger branch as input
+	pipeline := tu.UniqueString("pipeline")
+
+	txn, err := c.StartTransaction()
+	require.NoError(t, err)
+
+	require.NoError(t, c.WithTransaction(txn).CreatePipeline(
+		pipeline,
+		"",
+		[]string{"bash"},
+		[]string{
+			fmt.Sprintf("cp /pfs/%s/* /pfs/out/", dataRepo),
+		},
+		&pps.ParallelismSpec{
+			Constant: 1,
+		},
+		client.NewPFSInput(dataRepo, "/*"),
+		"",
+		false,
+	))
+
+	info, err := c.InspectTransaction(txn)
+	require.NoError(t, err)
+	require.Equal(t, 1, len(info.Responses))
+	require.NotNil(t, info.Responses[0].CreatePipelineResponse)
+	id := info.Responses[0].CreatePipelineResponse.FileSetId
+
+	// make sure the fileset is gone
+	// TODO: do this without a sleep (delete from database directly?)
+	require.NoError(t, c.RenewFileSet(id, time.Second))
```

Comment:
We support negative durations internally for this exact use-case.  Check if they work through the API, if not, this is fine for now.

Reply:
The API has an explicit check disallowing durations < 1 second, but even if I set it negative, it seems like reads of filesets succeed regardless of TTL, so I thought I'd need to wait for GC anyway.

Reply:
This is fine for now.  We can probably expose a way to force the GC to run.